app/src/main/java/com/github/onlynotesswent/model/users/UserRepositoryFirestore.kt

```diff
@@ -96,11 +97,32 @@ class UserRepositoryFirestore(private val db: FirebaseFirestore) : UserRepositor
   }
 
   override fun deleteUserById(id: String, onSuccess: () -> Unit, onFailure: (Exception) -> Unit) {
-    db.collection(collectionPath)
-        .document(id)
-        .delete()
-        .addOnSuccessListener { onSuccess() }
-        .addOnFailureListener { exception -> onFailure(exception) }
+    getUserById(
+        id,
+        { user ->
+          // for each of the current user follower remove them from their following list
```

Comment:
Isn't this the purpose of the function `removeFollowerFrom` ? Is there a reason to not use it?

Reply:
Yes but it's faster this way as I'm only doing one firestore request instead of doing one for each follower

Reply:
Okay, makes sense then